boto3/resources/factory.py

```diff
@@ -133,9 +133,10 @@ def load_from_definition(self, resource_name,
 
         base_classes = [ServiceResource]
         if self._emitter is not None:
-            self._emitter.emit('creating-resource-class.%s' % cls_name,
-                               class_attributes=attrs,
-                               base_classes=base_classes)
+            self._emitter.emit(
+                'creating-resource-class.%s' % cls_name,
+                class_attributes=attrs, base_classes=base_classes,
+                service_context=service_context, emitter=self._emitter)
```

Comment:
Service context I see makes sense.  I think it's odd to include the emitter in an event emitted by the emitter.

Given you need the emitter to register a function, why not just give the emitter to the class/function on registration instead of at event emission time?